File: src/main/java/me/itzisonn_/meazy/lang/TextException.java
package me.itzisonn_.meazy.lang;

import me.itzisonn_.meazy.lang.text.Text;
import org.jspecify.annotations.NullMarked;

/**
 * Exception that accepts {@link Text} instead of {@link String}
 */
@NullMarked
public abstract class TextException extends RuntimeException {
    /**
     * @param text Text
     */
    public TextException(Text text) {
        super(text.toString());
    }
}

File: src/main/java/me/itzisonn_/meazy/text/TextException.kt
package me.itzisonn_.meazy.text

/**
 * Exception that accepts [Text] instead of [String]
 *
 * @param text Text
 */
abstract class TextException @JvmOverloads constructor(
    text: Text,
    e: Throwable? = null
) : RuntimeException(text.toString(), e)
